admin/tool/policy/classes/api.php

```diff
@@ -137,6 +142,28 @@ public static function get_policy_version($policyid, $versionid) {
         return $DB->get_record_sql($sql, $params, MUST_EXIST);
     }
 
+    /**
+     * Calculate the status of a particular policy document version.
+     *
+     * @param int $policyid ID of the policy document.
+     * @param int $versionid ID of the policy document revision.
+     * @return string Status of the policy document version.
+     */
+    public static function get_policy_version_status($policyid, $versionid) {
+        $status = static::VERSION_STATUS_DRAFT;
+        $policy = static::list_policies($policyid)[$policyid];
```

Comment:
Why are we loading all policies and all their versions here to get a status of a single one? I would expect that this function takes an existing policy version record as the argument, and it focuses only on the status calculation. Let it do one thing, without side effects.